### ISSUE TITLE
Compare doesn't work between certain versions

### ISSUE THREAD

**Issue by aw** [aw/picolisp-semver#1]
A bug in the `(cmp-versions)` function returns the wrong result for a comparison between two versions, ex:

```
: (semver-compare "2.6.0" "3.0.0")
-> T
: (semver-compare "2.0.0" "3.0.0")
-> NIL

: (semver-compare-slow "2.6.0" "3.0.0")
-> NIL
: (semver-compare-slow "2.0.0" "3.0.0")
-> NIL
```

Works fine with `(semver-compare-slow)`

**Comment by aw** [comment]
Errr... still not fixed:

```
: (semver-compare "5.6.0" "2.6.0")
-> T
: (semver-compare "5.7.0" "2.6.0")
-> T
: (semver-compare "5.5.0" "2.6.0")
-> NIL
```

These should all return `T` since the major always is greater.

**Comment by aw** [comment]
Fixed in 4a852a9
